apiary-metastore-events/kafka-metastore-events/kafka-events-common/src/main/java/com/expediagroup/apiary/extensions/events/metastore/common/listener/WrappingMessageTask.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (C) 2018-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.apiary.extensions.events.metastore.common.listener;
+
+import static com.expediagroup.apiary.extensions.events.metastore.common.listener.ListenerUtils.error;
+import static com.expediagroup.apiary.extensions.events.metastore.common.listener.ListenerUtils.success;
+
+import java.util.Objects;
+
+import com.expediagroup.apiary.extensions.events.metastore.common.messaging.MessageTask;
+
+class WrappingMessageTask implements MessageTask {
```

Comment:
Why do we have this instead of just putting all this logic into `MessageTask` and having that be a concrete class instead of an interface? For testing maybe? (I appreciate finding the answers to some of this might require some code archaeology...)

Reply:
My guess is because in Shunting Yard we supported three `MessageTask` types (sqs, kinesis, kafka). They have different run logic, but the same logic to execute, so I guess this was extracted to remove code duplication.

In any case, this is part of the refactorings to come, so will leave this for now